sdk/python/kfp/dsl/_container_op.py

```diff
@@ -848,7 +850,7 @@ def inputs(self):
     # called the 1st time (because there are in-place updates to `PipelineParam`
     # during compilation - remove in-place updates for easier debugging?)
     if not self._inputs:
-      self._inputs = []
+      self._inputs = self._component_spec_inputs or []
       # TODO replace with proper k8s obj?
       for key in self.attrs_with_pipelineparams:
         self._inputs += _pipeline_param.extract_pipelineparams_from_any(
```

Comment:
Looks like `_component_spec_inputs` is a list of inputs reference `PipelineParam`, in another word, inputs with constant values would not appear in this list. It that right? 
If so, 1) maybe we can rename it to better reflect this fact. 2) I wonder if this `self._inputs += _pipeline_param.extract_pipelineparams_from_any` is still needed, because it looks to me you would have covered them when you construct `_component_spec_inputs` down below?

Reply:
Thanks for the explanation. That makes sense. 
(This also reminds me the topic that component may have implicit inputs not captured in its inputs spec, and whether we should allow this to happen)